lib/spack/spack/repo.py

```diff
@@ -1362,6 +1362,33 @@ def get_pkg_class(self, pkg_name):
         if not inspect.isclass(cls):
             tty.die("%s.%s is not a class" % (pkg_name, class_name))
 
+        new_cfg_settings = (
+            spack.config.get("packages").get(pkg_name, {}).get("package_attributes", {})
+        )
```

Comment:
At the moment no. I can look into how to prohibit this in the schema.

Reply:
Actually it occurred to me this could be useful for:

* `submodules = True` - a user might want to set this universally
* If we promote `full_git_repo` from a per-version option to a package-level option (like `submodules`)

It seems more-constrained though: like we ought to perhaps white-list entries for `all`.

Reply:
I think for now it's fine if we just document that you can't set it for `all` nor for a virtual